dask_ml/model_selection/_incremental.py

```diff
@@ -772,13 +773,17 @@ def inverse(time):
 
         current_time_step = time_step + 1
         next_time_step = current_time_step
-        while inverse(current_time_step) == inverse(next_time_step) and (
-            not self.patience
-            or next_time_step - current_time_step < self.scores_per_fit
-        ):
-            next_time_step += 1
+        if inverse(current_time_step) == 0:
+            # we'll never get out of here
+            next_time_step = 1
```

Comment:
Maybe the `inverse` function itself should never produce non-positive values?  I can imagine placing the `min(1, ` there instead, or perhaps using `math.ceil` rather than `int`

Reply:
I think this additional `if` check is still necessary, even if we change `inverse` to be always positive. When these start off at the same value (whether it's 0 or 1), we get stuck in a cycle where `next_time_step` is increasing, making the `inverse(current_time_step) == inverse(next_time_step)` condition even more false.